stl/inc/flat_set

```diff
@@ -621,32 +628,30 @@ private:
 
     void _Erase_dupes_if_not_multi() {
         if constexpr (!_Multi) {
-            const auto _Equal_to = [this](const _Kty& _Lhs, const _Kty& _Rhs) {
+            const auto _Equivalent = [this](const _Kty& _Lhs, const _Kty& _Rhs) {
                 return !_Compare(_Lhs, _Rhs) && !_Compare(_Rhs, _Lhs);
             };
-            const iterator _End     = end();
-            const iterator _New_end = _STD unique(begin(), _End, _Equal_to);
-            _Get_cont().erase(_New_end, _End);
+            const iterator _End = end();
+            _Mycont.erase(_STD unique(begin(), _End, _Equivalent), _End);
         }
     }
 
     template <bool _Presorted>
     void _Restore_invariants_after_insert(const size_type _Old_size) {
-        auto _Comp              = _Get_comp_v();
         const iterator _Begin   = begin();
         const iterator _Old_end = _Begin + static_cast<difference_type>(_Old_size);
-        const iterator _New_end = end();
+        const iterator _End     = end();
 
         if constexpr (!_Presorted) {
-            _STD sort(_Old_end, _New_end, _Comp);
+            _STD sort(_Old_end, _End, _Pass_comp());
         } else {
-            _STL_ASSERT(_Check_sorted(_Old_end, _New_end), _Msg_not_sorted);
+            _STL_ASSERT(_Is_sorted(_Old_end, _End), _Msg_not_sorted);
         }
 
-        _STD inplace_merge(_Begin, _Old_end, _New_end, _Comp);
+        _STD inplace_merge(_Begin, _Old_end, _End, _Pass_comp());
         _Erase_dupes_if_not_multi();
 
-        _STL_INTERNAL_CHECK(_Check_sorted(cbegin(), cend()));
+        _STL_INTERNAL_CHECK(_Is_sorted(cbegin(), cend()));

```

Comment:
regaining `cend` is necessary, as `_Erase_dupes_if_not_multi` may invalidate `end`. I think for better clarity we'd better use `_Is_sorted(_Mycont.cbegin(), _Mycont.cend())` when the check is container-scale.